packages/dataviews/src/components/dataviews-item-actions/index.tsx

```diff
@@ -106,18 +106,20 @@ export function ActionModal< Item >( {
 	const label =
 		typeof action.label === 'string' ? action.label : action.label( items );
 	return (
-		<Modal
-			title={ action.modalHeader || label }
-			__experimentalHideHeader={ !! action.hideModalHeader }
-			onRequestClose={ closeModal }
-			focusOnMount={ action.modalFocusOnMount ?? true }
-			size={ action.modalSize || 'medium' }
-			overlayClassName={ `dataviews-action-modal dataviews-action-modal__${ kebabCase(
-				action.id
-			) }` }
-		>
-			<action.RenderModal items={ items } closeModal={ closeModal } />
-		</Modal>
+		<Suspense fallback={ null }>
+			<Modal
```

Comment:
Perhaps you could argue that we could put `<Suspense>` boundaries within `<Modal>` so that every modal had the same behaviour.

If the design system had an official way we believed slow modals should load (perhaps an overlay with a spinner in the middle of it?) then I think we could justify that change.